### PR TITLE
⚡️ feat: TSify Sapper's input files (client.js, server.js, service-worker.js → .ts)

### PR DESCRIPTION
**Pull request by babichjacob**
fix linting banning // @ts-ignore, remove unnecessary usage of it, and explain every one that remains

### CODE REVIEW

**Comment by babichjacob** [comment]
This replaces #4 disabling TypeScript when not in use, because now there will always be `.ts` files in `src`.

File: package.json [hunk 1]
```diff
@@ -30,7 +30,7 @@
 		"compression": "^1.7.4",
 		"express": "^4.17.1",
 		"node-fetch": "^2.6.0",
-		"sirv": "^0.4.6"
+		"sirv": "^1.0.0-next.5"
```
Comment (benmccann) [inline package.json]: any reason you switched to the pre-release version?

File: src/service-worker.ts [hunk 2]
```diff
@@ -10,31 +9,31 @@ const ASSETS = `cache${timestamp}`;
 const toCache = shell.concat(files);
 const cached = new Set(toCache);
 
-self.addEventListener("install", (event) => {
+self.addEventListener("install", (event: any /* ExtendableEvent */) => {
```
Comment (benmccann) [inline src/service-worker.ts]: I think I'd played with this a little and had to add `webworker` to the `tsconfig`. I think was able to just do `event: ExtendableEvent` then

```
     "compilerOptions": {
          "lib": [
               "webworker",
```

File: package.json (hunk 1 above)
Comment (babichjacob) [reply package.json]: I expected types to be available in this version but they won't be until the next publish (hopefully).

Comment (babichjacob) [reply package.json]: Reverted in newest commit

File: src/service-worker.ts [hunk 3]
```diff
@@ -46,7 +46,9 @@ self.addEventListener("fetch", (event: any /* ExtendableEvent */) => {
 
 	// always serve static files and bundler-generated assets from cache
 	if (url.host === self.location.host && cached.has(url.pathname)) {
-		event.respondWith(caches.match(event.request));
+		caches.match(event.request).then((match): void => {
+			if (match) event.respondWith(match);
```
Comment (babichjacob) [inline src/service-worker.ts:50]: Are there side effects to letting the `event` never be responded to if there's no match?

By the way, this "solves" https://github.com/sveltejs/sapper/issues/760#issuecomment-632974392

File: src/service-worker.ts (hunk 2 above)
Comment (babichjacob) [reply src/service-worker.ts]: 653fd12 

🙏🏻 Thank you

File: src/service-worker.ts (hunk 3 above)
Comment (benmccann) [reply src/service-worker.ts:50]: I don't know. I haven't used service workers much. But I couldn't find any examples online that did anything else

**Pull request merged**